Homework3/Homework3/Program.cs

```diff
@@ -0,0 +1,238 @@
+
+            if (CheckIfCredentialsAreValid(filePath, userName, password))
+            {
+                Console.WriteLine($"Hello!{Environment.NewLine}");
+            }
+            else
+            {
+                Console.WriteLine($"Invalid credentials.{Environment.NewLine}");
+            }
+        }
+
+        /// <summary>
+        /// Prompts the user for input.
+        /// </summary>
+        /// <param name="inputType">The type of data being prompted for.</param>
+        static string PromptUserForInput(string inputType)
+        {
+            Console.Write($"Enter a {inputType}: ");
+            string input = Console.ReadLine();
+
+            return input;
+        }
+
+        /// <summary>
+        /// Checks the specified file to see if the credentials are valid.
+        /// </summary>
+        /// <param name="filePath">The file being checked against.</param>
+        /// <param name="userName">The user name being checked.</param>
+        /// <param name="password">The password being checked.</param>
+        /// <returns>
+        /// True if the user name and password match, false if it does not match.
+        /// </returns>
+        static bool CheckIfCredentialsAreValid(string filePath, string userName, string password)
+        {
+            StreamReader reader = new StreamReader(filePath);
+
+            bool isValid = false;
+
+            while (reader.Peek() >= 0)
+            {
+                string line = reader.ReadLine();
+                string[] words = line.Split(",");
+
+                isValid = words[0].Equals(userName) && words[1].Equals(password);
+
+                if (isValid)
+                {
+                    break;
+                }
+            }
+
+            reader.Close();
+
+            return isValid;
+        }
+
+        /// <summary>
+        /// Registers a new user and adds their user name and password 
+        /// to the specified file. If the user is a duplicate, try again.
+        /// </summary>
+        /// <param name="filePath">The path of the file being written to.</param>
+        static void RegisterUser(string filePath)
+        {
+            string userName;
+            string password;
+            bool isDuplicate;
+
+            do
+            {
+                userName = PromptUserForInput("user name");
+                password = PromptUserForInput("password");
+
+                try
+                {
+                    isDuplicate = CheckIfDuplicate(filePath, userName); 
+                }
+                catch (DuplicateUserCredentialsException)
+                {
+                    isDuplicate = true;
+                    Console.WriteLine($"User already exists.{Environment.NewLine}");
+                }
+
+                if (!isDuplicate)
+                {
+                    WriteToDataFile(filePath, userName, password);
+                    Console.WriteLine($"Credentials added.{Environment.NewLine}");
+                    break;
```

Comment:
Do you really need a break here? Won't the do..while loop stop anyways on this condition?